packages/xchain-ethereum/src/client.ts

```diff
@@ -570,16 +563,17 @@ export default class Client implements XChainClient, EthereumClient {
    * Thrown if the given contract address is empty.
    */
   estimateCall = async (
-    address: Address,
+    walletIndex = 0,
```

Comment:
For an estimate call; we don't need to have the wallet seed phrase

We can just do

```
contract.estimateGas[func](...params)
```

And add `from: senderAddress` 

```
ct for a read-only method may include any of:

    overrides.from - the msg.sender (or CALLER) to use during the execution of the code
    overrides.value - the msg.value (or CALLVALUE) to use during the exectuiont of the code
    overrides.gasPrice - the price to pay per gas (theoretically); since there is no transaction, there is not going to be any fee charged, but the EVM still requires a value to report to tx.gasprice (or GASPRICE); most developers will not require this
    overrides.gasLimit - the amount of gas (theoretically) to allow a node to use during the execution of the code; since there is no transaction, there is not going to be any fee charged, but the EVM still processes gas metering so calls like gasleft (or GAS) report meaningful values
    overrides.blockTag - a block tag to simulate the execution at, which can be used for hypothetical historic analysis; note that many backends do not support this, or may require paid plans to access as the node database storage and processing requirements are much higher 
    ```

cf. https://docs.ethers.io/v5/api/contract/contract/